src/ocean/application/components/GCStats.d

```diff
@@ -0,0 +1,216 @@
+/*******************************************************************************
+
+    Contains API to obtain garbage collector stats.
+
+    Copyright:
+        Copyright (c) 2009-2018 dunnhumby Germany GmbH.
+        All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.application.components.GCStats;
+
+import core.memory;
+
+import ocean.core.Traits : hasMember, ParameterTupleOf;
+import ocean.time.MicrosecondsClock;
+
+/*******************************************************************************
+
+    Class used for getting used GC stats for the current process. This will
+    work only with dmd-transitional.
+
+*******************************************************************************/
+
+public class GCStats
+{
+    static if ( hasMember!(GC, "monitor") )
+    {
+        /**********************************************************************
```

Comment:
General point: feel free to use `///` comments for field documentation in ocean. (But no need to change this PR now.)